jetty-client/src/main/java/org/eclipse/jetty/client/ValidatingConnectionPool.java

```diff
@@ -105,50 +95,28 @@ public boolean release(Connection connection)
     @Override
     public boolean remove(Connection connection)
     {
-        Holder holder;
-        lock();
-        try
-        {
-            holder = quarantine.remove(connection);
-        }
-        finally
-        {
-            unlock();
-        }
+        Holder holder = quarantine.remove(connection);
 
-        if (holder == null)
-            return super.remove(connection);
+        if (holder != null)
+            holder.cancel();
```

Comment:
I've refactored this code to be more like it previously was; and it seems the problem of calling `super.remove` twice was already present. But this doesn't have any ill-effect: it can be safely be called concurrently.